.github/workflows/all-static-checks.yaml

```diff
@@ -88,6 +88,48 @@ jobs:
         run: |
           pip install pyyaml
           python tests/sweep_framework/framework/sweeps_workflow_verification.py
+  cmake-version:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: recursive
+    - name: Check for changed CMake files
+      id: changed-cmake-files
+      uses: tj-actions/changed-files@v45
+      with:
+        files: |
+          **/*.cmake
+          **/CMakeLists.txt
+    - uses: lukka/get-cmake@v3.30.5
+      if: steps.changed-cmake-files.outputs.any_changed == 'true'
+      with:
+        cmakeVersion: "~3.16.0"
+    - name: Check CMake version
+      if: steps.changed-cmake-files.outputs.any_changed == 'true'
+      run: cmake --version
+    - name: Install LLVM and Clang
+      if: steps.changed-cmake-files.outputs.any_changed == 'true'
+      run: |
+        wget https://apt.llvm.org/llvm.sh
+        chmod u+x llvm.sh
+        sudo ./llvm.sh 17
+    - name: Install deps
+      if: steps.changed-cmake-files.outputs.any_changed == 'true'
+      env:
+        DEBIAN_FRONTEND: noninteractive
+      run: |
+        sudo apt update
+        sudo xargs -a scripts/docker/requirements-22.04.txt apt install -y --no-install-recommends
+    - name: Check CMake compatibility
+      if: steps.changed-cmake-files.outputs.any_changed == 'true'
+      env:
+        ARCH_NAME: wormhole_b0
```

Comment:
Hahahahah this sucks - can't wait until @blozano-tt is done his work

Reply:
It might take a village:
https://github.com/tenstorrent/tt-metal/milestone/13